docs/evaluation.md

```diff
@@ -1,82 +1,185 @@
+    results["passed"] = all([
+        results["token_efficient"],
+        results["step_efficient"],
+        results["has_output"]
+    ])
+
+    return results
+
+# Usage
 from any_agent import AgentConfig, AnyAgent
+from any_agent.evaluation import LlmJudge
 from any_agent.tools import search_web
 
+# First, run an agent to get a trace
 agent = AnyAgent.create(
-    "langchain",
+    "tinyagent",
     AgentConfig(
-        model_id="gpt-4o-mini",
+        model_id="gpt-4.1-nano",
         tools=[search_web]
     ),
 )
+trace = agent.run("What is the capital of France?")
+evaluation = evaluate_efficiency(trace)
+print(f"Evaluation results: {evaluation}")
+```
+
+### Working with Trace Messages
 
-agent_trace = agent.run("How many seconds would it take for a leopard at full speed to run through Pont des Arts?")
+You can also examine the conversation flow directly:
+
+```python
+def check_tool_usage(trace: AgentTrace, required_tool: str) -> bool:
+    """Check if a specific tool was used in the trace."""
+    messages = trace.spans_to_messages()
+
+    for message in messages:
+        if message.role == "tool" and required_tool in message.content:
+            return True
+    return False
+
+# Usage
+used_search = check_tool_usage(trace, "search_web")
+print(f"Used web search: {used_search}")
 ```
 
+## LlmJudge
+
+The `LlmJudge` is ideal for straightforward evaluation questions that can be answered by examining the complete trace text. It's efficient and works well for:
 
-### Define an evaluation case either in a yaml file or in python:
-
-=== "YAML"
-    ~~~yaml
-    {% include "./examples/evaluation_case.yaml" %}
-    ~~~
-    Then in python
-    ```python
-    from any_agent.evaluation.evaluation_case import EvaluationCase
-    evaluation_case = EvaluationCase.from_yaml(evaluation_case_path)
-    ```
-
-=== "Python"
-    ```python
-    from any_agent.evaluation.evaluation_case import EvaluationCase
-    evaluation_case = EvaluationCase(
-            ground_truth={ "value": "9", "points": 1.0},
-            checkpoints=[
-                {"criteria": "Did the agent run a calculation", "points": 1},
-                {"criteria": "Did the agent use fewer than 5 steps", "points": 4},
-            ],
-            llm_judge="gpt-4o-mini",
-    )
-    ```
-
-### Run the evaluation using the test case and trace
-
-=== "Python"
-    ```python
-    from any_agent.evaluation.evaluate import evaluate
-    eval_result = evaluate(
-        evaluation_case=evaluation_case,
-        trace=agent_trace
-    )
-    print(f"Final score: {eval_result.score}")
-    print(f"Checkpoint scores: {eval_result.checkpoint_results}")
-    ```
-
-=== "CLI"
-    ```bash
-    any-agent-evaluate \
-        --evaluation_case_path "docs/examples/evaluation_case.yaml" \
-        --trace_path "tests/assets/OPENAI_trace.json"
-    ```
+- Basic pass/fail assessments
+- Simple criteria checking
+- Text-based evaluations
+
+### Example: Basic LLM Judge
```

Comment:
Not a blocker: we should try to come up with a more real/useful example to showcase